### PR TITLE
Added week7 solutions

### PR DESCRIPTION
**Pull request by mdreaj38**

### CODE REVIEW

File: week07/meetingRoomII.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//TC: log(N),
```
Comment (Sadman007) [inline week07/meetingRoomII.cpp]: The correct TC is `O(NlogN)`

File: week07/randomPickWithWeight.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// TC: O(N)
```
Comment (Sadman007) [inline week07/randomPickWithWeight.cpp:1]: The pick index is `O(logN)` for the binary search.

File: week07/searchA2dMatrix.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    candidateRow = mid;
+				}
+		}
+        
+		return binary_search(matrix[candidateRow].begin(), matrix[candidateRow].end(), target);   
+    }
+
+	
+	// Approach 2
+	// TC: log(m+n)
```
Comment (Sadman007) [inline week07/searchA2dMatrix.cpp]: It is `M+N`, not `log(M+N)` brother :)

**Review by Sadman007 — APPROVED** [review]
Hi @mdreaj38 

1) Format your codes before pushing them into GitHub. Most of the codes are not readable.
2) Work hard on complexity analysis.

File: week07/randomPickWithWeight.cpp (hunk 2 above)
Comment (mdreaj38) [reply week07/randomPickWithWeight.cpp:1]: Vaia I sorted the array at line no 6. Thats why overall complexity is NlogN here

**Pull request merged**